usr/lib/ica_s390_stdll/ica_specific.c

```diff
@@ -4639,6 +5019,10 @@ static CK_RV mech_list_ica_initialize(STDLL_TokData_t *tokdata)
         if (libica_func_list[i].mech_mode_id == AES_CBC)
             ica_data->ica_aes_available = TRUE;
 
+        /* Remember if libica supports the new AES-GCM API (z14 and later) */
+        if (libica_func_list[i].mech_mode_id == AES_GCM_KMA)
```

Comment:
Can it be that AES_GCM_KMA is available but AES_CBC is not available ?
If that can happen, then even if AES_GCM_KMA is available, it will always take the OpenSSL software fallback, because AES_CBC is not available.

Reply:
Hm, I would say no. But we could put
```
if (ica_data->ica_new_gcm_available)
        return new_gcm_specific_aes_gcm_update/final(...)
```
before 
```
if (!ica_data->ica_aes_available)
        return openssl_specific_aes_gcm_update/final(...)
```
But in the token_specific_aes_gcm_init routine we would then also have to put 
```
if (!ica_data->ica_aes_available)
        return openssl_specific_aes_gcm_init(...);
```
down, after getting the key value. What do you think? On a messed up system with KMA available, but CBC not, we would then use openssl. Would that matter?

Reply:
Hm, no, leave it as is.